tests/items/cache_test.py

```diff
@@ -0,0 +1,783 @@
+        cached_result1 = test_file.cache.get_dict_cache(True)
+        cached_result2 = test_file.cache.get_dict_cache(False)
+
+    # Assert
+    assert (cached_result1 == result1) == expected_output
+    assert (cached_result2 == result2) == expected_output
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_exception,expected_output",
+    [
+        (True, does_not_raise(), True),
+        (False, does_not_raise(), False),
+    ],
+)
+def test_mgmtclass_dict_cache_load(
+    cobbler_api, input_value, expected_exception, expected_output
+):
+    # Arrange
+    cobbler_api.settings().cache_enabled = input_value
+    test_mgmtclass = Mgmtclass(cobbler_api)
+    test_mgmtclass.name = "test_mgmtclass"
+    cobbler_api.add_mgmtclass(test_mgmtclass)
+
+    # Act
+    with expected_exception:
+        result1 = test_mgmtclass.to_dict(resolved=True)
+        result2 = test_mgmtclass.to_dict(resolved=False)
+        cached_result1 = test_mgmtclass.cache.get_dict_cache(True)
+        cached_result2 = test_mgmtclass.cache.get_dict_cache(False)
+
+    # Assert
+    assert (cached_result1 == result1) == expected_output
+    assert (cached_result2 == result2) == expected_output
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_exception,expected_output",
+    [
+        (True, does_not_raise(), True),
+        (False, does_not_raise(), False),
+    ],
+)
+def test_repo_dict_cache_load(
+    cobbler_api, input_value, expected_exception, expected_output
+):
+    # Arrange
+    cobbler_api.settings().cache_enabled = input_value
+    test_repo = Repo(cobbler_api)
+    test_repo.name = "test_repo"
+    cobbler_api.add_repo(test_repo)
+
+    # Act
+    with expected_exception:
+        result1 = test_repo.to_dict(resolved=True)
+        result2 = test_repo.to_dict(resolved=False)
+        cached_result1 = test_repo.cache.get_dict_cache(True)
+        cached_result2 = test_repo.cache.get_dict_cache(False)
+
+    # Assert
+    assert (cached_result1 == result1) == expected_output
+    assert (cached_result2 == result2) == expected_output
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_exception,expected_output",
+    [
+        (True, does_not_raise(), True),
+        (False, does_not_raise(), False),
+    ],
+)
+def test_distro_dict_cache_load(
+    cobbler_api, create_distro, input_value, expected_exception, expected_output
+):
+    # Arrange
+    cobbler_api.settings().cache_enabled = input_value
+    test_distro = create_distro()
+
+    # Act
+    with expected_exception:
+        result1 = test_distro.to_dict(resolved=True)
+        result2 = test_distro.to_dict(resolved=False)
+        cached_result1 = test_distro.cache.get_dict_cache(True)
+        cached_result2 = test_distro.cache.get_dict_cache(False)
+
+    # Assert
+    assert (cached_result1 == result1) == expected_output
+    assert (cached_result2 == result2) == expected_output
+
+
+@pytest.mark.skip("RuntimeError: dictionary changed size during iteration")
```

Comment:
This one is bad. If you can't find it, I will take the time and fix it.

Reply:
I could not find the cause of this error and would be grateful for help in resolving it.

Reply:
fixed in https://github.com/cobbler/cobbler/pull/3266/commits/c298938a8073e386279c21b59d0406d794ff2d11